pkg/operator/ceph/cluster/mgr/mgr.go

```diff
@@ -151,6 +152,17 @@ func (c *Cluster) Start() error {
 			}
 			logger.Infof("deployment for mgr %s already exists. updating if needed", resourceName)
 
+			mgrsToSkipReconcile, err := c.getMGRsToSkipReconcile()
+
+			if err != nil {
+				return errors.Wrap(err, "failed to check for mgrs to skip reconcile")
+			}
+
+			if mgrsToSkipReconcile.Len() > 0 {
```

Comment:
>> Instead of skipping if any are labeled, it would be ideal to skip only the ones labeled. The OSDs follow this pattern of only skipping the labeled ones. The mons just don't follow that pattern, but I think it's best for all other daemons.

> I think it would be ok to follow the same pattern like mon for mgr, mds and rgw. User might not want to upgrade the standby mds while debugging active mds

If they want to skip more than one daemon, they can label all the daemons they want to skip. If it's MDS, they could label both the active and corresponding standby daemons. For the mgr, they could label all the mgr deployments they want to skip. I still think that it will be most flexible if we treat the daemons independently for these skip labels. 

For this specific mgr check, I'd recommend moving the call for `c.getMGRsToSkipReconcile()` up above the `for` loop so it's only executed once for the reconcile. Then we could do this:
```go
if mgrsToSkipReconcile.Contains(daemonID) {
  logger.Warningf("Skipping reconcile of mgr %q since labeled with %s", daemonID, cephv1.SkipReconcileLabelKey)
  continue
}
```